core/src/main/java/io/grpc/internal/AbstractClientStream.java

```diff
@@ -95,7 +106,7 @@ protected void inboundTransportError(Status errorStatus) {
     }
     // For transport errors we immediately report status to the application layer
     // and do not wait for additional payloads.
-    transportReportStatus(errorStatus, false, new Metadata());
+    transportReportStatus(errorStatus, false, metadata);
   }
 
   /**
```

Comment:
Done, with verification in test. That left only one call-site for the 1-arg version of transportReportStatus, so I converted it to the 2-arg version and removed the 1-arg version of the method.